homeassistant/components/anomaly/binary_sensor.py

```diff
@@ -0,0 +1,326 @@
+                entity_id=entity_id,
+                attribute=attribute,
+                device_class=device_class,
+                invert=invert,
+                require_both=require_both,
+                positive_only=positive_only,
+                negative_only=negative_only,
+                max_samples=max_samples,
+                sample_duration=sample_duration,
+                max_trailing_samples=max_trailing_samples,
+                trailing_sample_duration=trailing_sample_duration,
+                min_change_amount=min_change_amount,
+                min_change_percent=min_change_percent,
+            )
+        )
+    if sensors:
+        add_entities(sensors)
+
+
+class SensorAnomaly(BinarySensorEntity):
+    """Representation of an anomaly Sensor."""
+
+    def __init__(
+        self,
+        hass,
+        device_id: str,
+        friendly_name: str,
+        entity_id: str,
+        attribute: str,
+        device_class: str,
+        invert: bool,
+        require_both: bool,
+        positive_only: bool,
+        negative_only: bool,
+        max_samples: int,
+        sample_duration: int,
+        max_trailing_samples: int,
+        trailing_sample_duration: int,
+        min_change_amount: float,
+        min_change_percent: float,
+    ):
+        """Initialize the sensor."""
+        self._hass = hass
+        self.entity_id: str = generate_entity_id(ENTITY_ID_FORMAT, device_id, hass=hass)
+        self._name: str = friendly_name
+        self._entity_id: str = entity_id
+        self._attribute: str = attribute
+        self._device_class: str = device_class
+        self._invert: bool = invert
+        self._require_both: bool = require_both
+        self._positive_only: bool = positive_only
+        self._negative_only: bool = negative_only
+        self._max_samples: int = max_samples
+        self._sample_duration: int = sample_duration
+        self._max_trailing_samples: int = max_trailing_samples
+        self._trailing_sample_duration: int = trailing_sample_duration
+        self._min_change_amount: float = min_change_amount
+        self._min_change_pct: float = min_change_percent
+        self._change_amount: float = 0.0
+        self._change_percent: float = 0.0
+        self._trailing_avg: float = 0.0
+        self._sample_avg: float = 0.0
+        self._state = None
+        self.trailing_samples: deque = deque(maxlen=max_trailing_samples)
+        self.samples: deque = deque(maxlen=max_samples)
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def is_on(self) -> str:
+        """Return true if sensor is on."""
+        return self._state
+
+    @property
+    def device_class(self) -> str:
+        """Return the sensor class of the sensor."""
+        return self._device_class
+
+    @property
+    def device_state_attributes(self) -> dict:
+        """Return the state attributes of the sensor."""
```

Comment:
We can probably just expose those attributes that are dynamic and skip those defined in the configuration

Reply:
Removed the non dynamic ones and committed it.